website/server/libs/cron.js

```diff
@@ -277,17 +305,7 @@ export function cron (options = {}) {
   // Tally each task
   let todoTally = 0;
 
-  tasksByType.todos.forEach(task => { // make uncompleted To-Dos redder (further incentive to complete them)
-    scoreTask({
-      task,
-      user,
-      direction: 'down',
-      cron: true,
-      times: multiDaysCountAsOneDay ? 1 : daysMissed,
-    });
-
-    todoTally += task.value;
-  });
+  todoTally = ageTodos(user, tasksByType, multiDaysCountAsOneDay, daysMissed, todoTally);
```

Comment:
I think this `todoTally` variable and the code below that uses it is also important to do while the user is sleeping. Since we're changing task values now while sleeping, we'd want the tallied history data to be adjusted too.

That would mean copying yet another thing into the `performSleepTasks` function.  There's also a bunch of other things that we really should be doing when the user is sleeping, such as tallying experience (`expTally`), the `tasksOrder` code around line 418 (there'll still be tasksOrder bugs for any user who's sleeping!), `preenUserHistory`, and probably almost everything else except actually doing damage from Dailies.

Can we consider getting rid of `performSleepTasks` and just putting an `if !sleeping` condition around the few pieces of code that should not be executed while sleeping? Although I realise this is getting away from the point of the issue!

If we keep using `performSleepTasks` for now, this comment in website/server/libs/cron.js at line 426 will need to be changed in this PR:
_// TODO also do while resting in the inn. Note that later we'll be allowing the value/color of tasks to change while sleeping (https://github.com/HabitRPG/habitica/issues/5232), so the code in performSleepTasks() might be best merged back into here for that. Perhaps wait until then to do preen history for sleeping users._

It could be changed to something like this:
_// TODO also run preenUserHistory while resting in the inn._

Reply:
Okay :P How about we map out some Cron flow offline and I'll just do some refactoring while allow these to happen. There are a few more conditions here, and I'd like to prevent adding hacking flow logic.